cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1218,8 +1218,11 @@ private void getRuns(HttpResponder responder, ProgramId programId, String status
     try {
       ProgramRunStatus runStatus = (status == null) ? ProgramRunStatus.ALL :
         ProgramRunStatus.valueOf(status.toUpperCase());
-      List<RunRecord> records =
-        Lists.transform(store.getRuns(programId, runStatus, start, end, limit), CONVERT_TO_RUN_RECORD);
+
+      Map<ProgramRunId, RunRecordMeta> recordMap = store.getRuns(programId, runStatus, start, end, limit);
+      List<RunRecordMeta> metaRecords = new ArrayList(recordMap.values());
```

Comment:
Why need to copy the values to a new list? I think we can use `Collections2.transform` on the map values directly and send back the result as Json (underneath Gson is used to do the serialization and it should be able to handle `Collection`).